x/evm/precompile/funtoken.go

```diff
@@ -151,7 +146,12 @@ func (p precompileFunToken) bankSend(
 			return
 		}
 	} else {
-		err = SafeMintCoins(ctx, evm.ModuleName, coinToSend, p.bankKeeper, start.StateDB)
+		// NOTE: The NibiruBankKeeper needs to reference the current [vm.StateDB] before
+		// any operation that has the potential to use Bank send methods. This will
+		// guarantee that [evmkeeper.Keeper.SetAccBalance] journal changes are
+		// recorded if wei (NIBI) is transferred.
+		p.evmKeeper.Bank.StateDB = start.StateDB
```

Comment:
Didn't we already set `p.evmKeeper.Bank.StateDB` in the `Run()` method, prior to calling `bankSend`? I think it's better to keep it in the `Run()` method because that executes prior to any specific method call.

Reply:
It gets messed up by the ERC20 tranfers and stuff

Reply:
It ends up creating a new reference to the *statedb.StateDB entirely. You'll find that if you remove these reassignments, it actually regresses and reintroduces the security vulnerability, failing our unit tests.

This test is the meaningful one:  `TestNativeSendThenPrecompileSend`

This issue arises from both `ERC20().Transfer` and `ERC20().Burn` calling ApplyEvmMsg with commit=true. This results in calling 
```go
	stateDB := k.NewStateDB(ctx, txConfig)
```
and creates a fresh pointer that differs from the parent StateDB that the precompile uses. Sadly, editing the child StateDB's journal doesn't affect the parent. 

And yes, adding the pointer assignment is also inexpensive